lib/OpenCloud/ObjectStore/Resource/Container.php

```diff
@@ -236,6 +236,28 @@ public function deleteAllObjects()
     }
 
     /**
+     * Delete an object from the API.
+     *
+     * @param string $name The name of object you want to delete
+     * @return True if object was delete with success. False if an error occurred
+     * @throws ObjectNotFoundException When the file you want to delete is not found
+     */
+    public function deleteObject($name)
+    {
+        try {
+            $response = $this->getClient()
+                    ->delete($this->getUrl($name))
+                    ->send();
+            return ($response->getStatusCode() == 204);
+        } catch (BadResponseException $e) {
+            if ($e->getResponse()->getStatusCode() == 404) {
+                throw ObjectNotFoundException::factory($name, $e);
+            }
+        }
+        return false;
+    }
```

Comment:
Yes it will be more consistent with the rest of API.
If the code change, we need to update unit test of this function.